submissions/factorio_draftsman/main.md

```diff
@@ -0,0 +1,212 @@
+
+The ROM is *very* dense, but it functions on a system where each value is split into and stored as two 16-bit numbers, which are then recombined on output. ROMs are normally tedious to create, manually encoding each signal you want by hand, one at a time, but this design was even more involved. I now had to split, bitwise AND, bitwise right-shift, and populate not one signal, but two, in two different specific locations, making sure that they both had the correct value *and* signal type. Needless to say, it was going to be painful to set tens of values, let alone hundreds, or the *thousands* the machine was capable of storing.
+
+The solution? Get a computer to do it for me. They can stomach this task far better than I ever could, and can do it way faster as well. Factorio's blueprint string import function can take any correctly formatted string; all I had to do was create this text string to my specification with the data I wanted and I could just paste it where I needed it.
+
+This concept is not new, at all. Even a cursory search will turn up numerous examples of this used in practice: [`factorio-blueprint` NPM module by demipixel](https://github.com/demipixel/factorio-blueprint), [justarandomgeek's compiler for his massive combinator computer](https://github.com/justarandomgeek/factorio-computer), [a generic combinator instruction language by Jobarion](https://github.com/Jobarion/factoriogen/), [image-to-blueprint converters](https://github.com/EFLFE/Artorio)... The list is extensive.
+
+With all these examples, I was hoping that I could find some similarities with some other repository and use that as a baseline for my solution to avoid "retreading". Something troubled me about the solutions that were available though. They all had problems! 
+
+* Many implementations were highly specific to the domain they were written for; a combinator computer compiler script was not going to be portable for many other purposes than compiling code for a specific combinator computer.
+* They lacked a unified language; many were written in Lua, some in Python, another in JavaScript, one in C++, etc. This meant that each one had to write it's own implementations for the same operations, instead of just having one guy write the implementation they need and making it available to others.
+* Many of them were written for versions of Factorio now severely out of date.
+* Documentation for a lot of these modules was sparse and sporadic, which turns off users like me who want to know what the module is capable of before investing any time learning how to use it.
+
+Fundamentally unsatisfied with the options, I resigned myself to my fate. I was going to have to follow in the footsteps of all before me, and develop my own implementation from scratch. How arduous!
+
+I made a prototype script in an afternoon that did exactly what I needed and it worked perfectly. It used a template system and it took less than a week to completely finish.
```

Comment:
Less than a week to finish your combinator computer, right?

Reply:
Well, more the encoding script for that particular ROM. Mk1 of the computer had already been "finished" by this point, and that took several months. Should that be made more clear?